diesel/src/query_source/peano_numbers.rs

```diff
@@ -0,0 +1,44 @@
+//! A simple implementation of peano numbers.
+//!
+//! This is used to enforce that columns can only be selected from a given from
+//! clause if their table appears exactly one time.
+
+/// A table never appears in the from clause.
+#[allow(missing_debug_implementations, missing_copy_implementations)]
```

Comment:
Are there any particular reasons why this should not implement `Debug`, `Clone` and `Copy`?

Reply:
Because instances of this type never exist, and more code = slower build times